sdk/download.go

```diff
@@ -151,3 +154,106 @@ func DownloadSDK(version string) error {
 	tarURL := TarballURL(version)
 	return DownloadSignedFile(tarFile, tarURL)
 }
+
+func fileSum(file string) ([]byte, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	hash := sha512.New()
+	if _, err := io.Copy(hash, f); err != nil {
+		return nil, err
+	}
+
+	return hash.Sum(nil), nil
+}
+
+func compareFileBytes(fileName1, fileName2 string) (bool, error) {
+	sum1, err := fileSum(fileName1)
+	if err != nil {
+		return false, err
+	}
+	sum2, err := fileSum(fileName2)
+	if err != nil {
+		return false, err
+	}
+	return bytes.Equal(sum1, sum2), nil
+}
+
+// UpdateFile downloads a file to temp dir and replaces the file only if
+// contents have changed. If tempDir is "" default will be os.TempDir().
+func UpdateFile(file, url, tempDir string) error {
+	t, err := ioutil.TempFile(tempDir, "sdkUpdateCheck")
+	if err != nil {
+		return nil
+	}
+	t.Close()
+	tempFile := t.Name()
+	defer os.Remove(tempFile)
+
+	if err := DownloadFile(tempFile, url); err != nil {
```

Comment:
Somewhat related:
https://github.com/vcaputo/coreos-install/blob/basic_installation/src/main.go#L206

We're doing quite similar things here, should probably share a package.